### PR TITLE
💥 🚚 Upgrade to Polymer 3

### PR DESCRIPTION
**Pull request by jordanfinners**
* Update to npm packages
* Update to es6 style
* Fix demo's and tests

Question @motss 
`<script src="/node_modules/web-animations-js/web-animations-next-lite.min.js"></script>`
`<script src="/node_modules/intl/dist/Intl.complete.js"></script>`
Need these to polyfill the browser so the animations and intl work, but can't es6 import them.
Do you think we should add them to the document head when initialising the element or just make users add them themselves when using the component?

---
This change is [Reviewable](https://reviewable.io/reviews/motss/app-datepicker/125)

### CODE REVIEW

File: app-datepicker-dialog.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      invalidDate: {
+        type: Boolean,
+        notify: true,
+        readOnly: true
+      },
+      inputDate: String,
+      showLongDate: Boolean,
+      noAnimation: Boolean,
+      pageEntryAnimation: String,
+      pageExitAnimation: String,
```
Comment (motss) [inline app-datepicker-dialog.js]: If `@polymer/neon-animation` will be removed as part of my suggestions in the comments, `page*Animation` property will no longer be needed.

File: app-datepicker.js [hunk 2]
```diff
@@ -0,0 +1,1756 @@
...
+        // https://www.polymer-project.org/1.0/docs/devguide/styling.html
+        this._updateDistributedButtonInkColorCustomProp(effectiveChildren[i], '#737373');
+      }
+      // update to a new height for datepicker if paper-buttons present.
+      this.$.dp.classList.add('with-buttons');
+    } else {
+      this.$.dp.classList.remove('with-buttons');
+    }
+
+    this.fire('app-datepicker-attached');
```
Comment (motss) [inline app-datepicker.js]: This inherits the old `attached` callback from WC v0. This should be rename to `app-datepicker-connected`.

File: demo/demo.html [hunk 3]
```diff
@@ -1,48 +1,53 @@
...
+  <script src="/node_modules/intl/dist/Intl.complete.js"></script>
+
+  <script>
+    window.Polymer = { lazyRegister: 'max', useNativeCSSProperties: !0, dom: 'shadow' };
+
+    // load webcomponents polyfills
+    (function () {
+      if ('registerElement' in document
+        && 'import' in document.createElement('link')
+        && 'content' in document.createElement('template')) {
```
Comment (motss) [inline demo/demo.html]: This needs to be updated as well. This is only valid for when using WC v0.

Use `web-components-loader` instead.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  "dependencies": {
+    "@polymer/iron-iconset-svg": "^3.0.0-pre.18",
+    "@polymer/iron-list": "^3.0.0-pre.18",
+    "@polymer/iron-selector": "^3.0.0-pre.18",
+    "@polymer/neon-animation": "^3.0.0-pre.18",
+    "@polymer/paper-dialog-behavior": "^3.0.0-pre.18",
+    "@polymer/paper-icon-button": "^3.0.0-pre.18",
+    "@polymer/polymer": "^3.0.0",
+    "web-animations-js": "^2.3.1",
+    "@polymer/paper-button": "^3.0.0-pre.18"
```
Comment (motss) [inline package.json]: All Polymer 3 based components should be out of beta. Please upgrade those to the latest.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    "@polymer/paper-button": "^3.0.0-pre.18",
+    "@polymer/paper-checkbox": "^3.0.0-pre.18",
+    "@polymer/paper-input": "^3.0.0-pre.18",
+    "@polymer/paper-dialog": "^3.0.0-pre.18",
+    "wct-browser-legacy": "^1.0.1",
+    "@polymer/iron-test-helpers": "^3.0.0-pre.18",
+    "web-animations-js": "^2.3.1",
+    "@webcomponents/webcomponentsjs": "^2.0.0",
+    "whatwg-fetch": "^2.0.0",
+    "intl": "^1.2.0"
```
Comment (motss) [inline package.json]: Please upgrade old dependencies here as well

File: .travis.yml [hunk 6]
```diff
@@ -1,16 +0,0 @@
...
-addons:
-  firefox: latest
-  apt:
-    sources:
-      - google-chrome
-    packages:
-      - google-chrome-stable
-script:
-  - xvfb-run polymer test
-dist: trusty
```
Comment (motss) [inline .travis.yml]: Is `.travis.yml` no longer needed?

**Comment by jordanfinners** [comment]
Will update the readme to reflect the optional importing of these dependencies - https://github.com/motss/app-datepicker/pull/125#issuecomment-443475080

For this comment - https://github.com/motss/app-datepicker/pull/125#issuecomment-443475212
All 3 I think are sensible, however I think that would be expanding this (already large) PR. Perhaps it could be done in steps rather than doing it all in one go in this piece of work?

Will work through your other PR comments and Codacy/codebeat reports 👍

File: .travis.yml (hunk 6 above)
Comment (jordanfinners) [reply .travis.yml]: I couldn't see it being used by anything, so removed it. Do you want it back in?

**Comment by jordanfinners** [comment]
Also will you be putting this onto npm once this PR is merged? As can't `bower install` it anymore once this is merged and the README will need updating.

And how did you test all the browser compatibility?

**Comment by motss** [comment]
Cool. Let's move those comments into another project. We're going to use `npm` starting Polymer 3. I'll publish the element once this is merged. BTW, the warnings and errors can be ignored from the `codebeat`.

**Comment by motss** [comment]
We rely on Travis for the cross browsers teatings. I decided to add Saucelabs back again which was removed earlier.

File: .travis.yml (hunk 6 above)
Comment (jordanfinners) [reply .travis.yml]: Ah okay, will add it back 👍

**Comment by jordanfinners** [comment]
@motss updated this PR with your feedback and some of the code check feedback.
 Are there any other PR's you want to get merged in before this one?

**Comment by jordanfinners** [comment]
Sounds good! :D 
The Gitmoji Police checker lies 😂

**Comment by motss** [comment]
@jordanfinners Thanks for your contribution. BTW, I noticed that you forced-push a commit for some reasons. Please don't do that as we need to find out what you have changed in each commit. In such way, we don't have to read every single changes and not knowing what you've made subsequently.

LGTM, I'll proceed to merging this PR!

**Comment by jordanfinners** [comment]
Thanks @motss !

Yeah I rebase my commits down to just one commit and force pushed it, just to keep it tidy :)

**Comment by jordanfinners** [comment]
Let me know when you've got it on NPM! :D

**Comment by motss** [comment]
@jordanfinners Not so soon to NPM publishing. I'm working on the tests and it is broken on some browsers. Feel free to help fixing them.

**Comment by jordanfinners** [comment]
Okay @motss, let me know what's broke and we can split it out between us?

**Comment by motss** [comment]
Test case `test for update to reflect external date change » should show correct output date when 'showLongDate' is falsy` is broken on Safari 9 - 12

**Comment by jordanfinners** [comment]
Ah I don't have Safari to help you on that one. Is there anything else you'd like testing/help on?

**Comment by motss** [comment]
Same here. I'm able to test on Macbook a couple days later in this week. I
don't mind NPM publishing it first as alpha version in the meantime.

**Comment by jordanfinners** [comment]
Yeah an alpha version on npm would be good! Let me know if you need any help with anything else! :)

**Comment by motss** [comment]
Published! https://www.npmjs.com/package/app-datepicker

**Comment by jordanfinners** [comment]
Does this issue need closing? (I checked but don't think I can 🙈) https://github.com/motss/app-datepicker/issues/112

Thank you for getting this through so quickly!

**Comment by motss** [comment]
I already closed the issue. Thanks for the effort for making this happen. Do let me know if the datepicker is working.

**Pull request merged**